src/main/java/frc/robot/subsystems/Led.java

```diff
@@ -0,0 +1,108 @@
+package frc.robot.subsystems;
+
+import org.littletonrobotics.junction.Logger;
+
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.maps.subsystems.LedMap;
+
+public class Led extends SmartSubsystemBase {
+
+    private LedMap map;
+    private Led defaultColor;
+    private int R;
+    private int G;
+    private int B;
+    AddressableLED led;
+    // Reuse buffer
+    // Default to a length of 60, start empty output
+    // Length is expensive to set, so only set it once, then just update data
+    AddressableLEDBuffer ledBuffer;
+
+    public Led(LedMap map) {
+        led = map.led;
+        ledBuffer = map.ledBuffer;
+
+        this.map = map;
+        // PWM port 9
+        // Must be a PWM header, not MXP or DIO
+
+        led.setLength(ledBuffer.getLength());
+
+        // Set the data
+        led.setData(ledBuffer);
```

Comment:
This still sketches me out that it's *after* the for loop setting the variables...